### PR TITLE
Fix issue #11396 : Rlist hides standard list constructors cons and nil

### LINKED ISSUE #11396: Rlist hides standard list constructors cons and nil in 8.11
In 8.11 we have this:
```
Require Import Reals.
Print cons.

Inductive Rlist : Set := nil : Rlist | cons : R -> Rlist -> Rlist
Arguments cons _%R_scope
```
This results in very substantial trouble in many modules which include Reals. A major issue is that many tactics use the standard list constructors. Unlike for Gallina code Coq cannot check at compile time if using such tactics would actually work - usually it doesn't. Many modules using Reals compile fine (possibly after some patching) but tactics don't work. The required patching can be substantial and tactics must be tested.

Since this is such a mess and likely also breaks a lot of user code (likely pretty much everything whic uses Reals) I would consider replacing Rlist with list R. I had a discussion with @ybertot on gitter on the topic and he mentioned that the raison d'être of Rlist is long gone (list was once restricted to Set).

I think we need to decide if we replace Rlist with list R and fix up the standard library in 8.11. I would strongly vote for it. It is likely faster than fixing up all the remaining failing modules in the 8.11 Windows installer (gappa is failing and proper picking for interval is extremely time consuming, some things are still untested).

### PR DESCRIPTION
**Pull request by MSoegtropIMC**
This PR fixes the hiding of the standard list cons and nil constructors by Rlist cons and nil when Reals is imported. This is done by not exporting Rlist from Reals and instead importing it (in the two places) where it is needed as suggested by @ppedrot.

Fixes #11396

### CODE REVIEW

**Comment by MSoegtropIMC** [comment]
Note: I tested locally that this builds, but I didn't test as yet if it fixes the issues I see. This will take until today evening.

**Review by ppedrot — CHANGES_REQUESTED** [review]
This needs a changelog before merging, but let's first run the CI to get an idea of the breakage.

**Comment by ppedrot** [comment]
So far, only coquelicot looks broken. I wrote a 3-line patch that makes it compile, and which is most likely backwards compatible. I'll submit a PR there.

**Comment by ppedrot** [comment]
See the [gitlab.inria.fr merge request](https://gitlab.inria.fr/coquelicot/coquelicot/merge_requests/2).

**Comment by ppedrot** [comment]
AFAIU, fiat-crypto failure is unrelated.

**Comment by Zimmi48** [comment]
Yeah, seems broken on master too.

**Comment by MSoegtropIMC** [comment]
@ppedrot : thanks for fixing Coquelicot! I will see tomorrow morning if Gappa works now and adjust the patch for Interval (90% of the patch can be removed now) and see if I can find a better pick for interval (which is much quicker with a smaller patch).

**Comment by MSoegtropIMC** [comment]
I did my best to write a changelog entry looking at others and also using the provies script, but somehow I messed it up. Can someone have a look and tell me what is wrong?

@ppedrot : shall I redirect Coquelicot to your fork/branch in the picking CI or shall I wait for upstream?

**Comment by ppedrot** [comment]
I'll try to get the upstream PR merged, but if that takes too long you can probably pick my repo instead. Let's see whether @silene is more responsive on GitHub.

**Comment by ppedrot** [comment]
The CI failures look somewhat spurious btw, maybe a rebase would suffice.

**Comment by MSoegtropIMC** [comment]
@ppedrot : my assumption is that I messed up the changelog entry. Most failing jobs ore doc jobs. Can you please have a look and see what I did wrong?

**Comment by ppedrot** [comment]
At first sight, the changelog does look correct though.

**Comment by ppedrot** [comment]
Ah, maybe it's missing a newline.

**Comment by Zimmi48** [comment]
Right, this is what says the linter, and all failing jobs that build the doc. The only other failure is Coquelicot.

**Comment by MSoegtropIMC** [comment]
OK, I will fix that.

**Comment by ejgallego** [comment]
As I suggested on Gitter, there is no need to wait for Coquelicot to apply the trivial patch if you folks want to be done with 8.11 [cc @ppedrot ] , you can just point the CI variable to the merge request and once the patch arrives upstream we undo.

The way I see CI overlays is pretty dynamic, at least for trivial patches like this.

**Comment by Zimmi48** [comment]
So FTR, the file to modify is `dev/ci/ci-basic-overlays.sh`.

**Comment by MSoegtropIMC** [comment]
@ejgallego : yes, I will do this in the picking PR if everything else in finished and this is not yet in upstream.

**Comment by Zimmi48** [comment]
You will also need to do it here so that CI is not broken on master...

**Comment by ejgallego** [comment]
Yes it has to be done here, I may push to the PR doing that and fixing the trivial mistake as not to delay the release more.

**Comment by Zimmi48** [comment]
This will need at least one approval from another member of @coq/reals-library-maintainers.

**Review by Zimmi48 — APPROVED** [review]

**Review by herbelin — APPROVED** [review]
> This will need at least one approval from another member of @coq/reals-library-maintainers.

This seems indeed to be the natural way to go.

**Comment by herbelin** [comment]
Actually, #11404 is proposing another solution, namely to just redefine `Rlist` from `list`. It is bit more radical but this is factorization which makes sense to me.

**Comment by Zimmi48** [comment]
This one is the short-term fix.

**Comment by MSoegtropIMC** [comment]
OK, I will do it here and cherry pick it for the picking PR tomorrow. Since Gappa is working now all which is left is a bit of cleanup and maybe an improvement for the picking of interval.

**Comment by ybertot** [comment]
I confirm that PR #11404 is intended to describe a possible long term solution, leading to more streamlined standard libraries.  I made the pull request to make sure ci would tell me how much damage it would make to depending packages.

**Review by ppedrot — APPROVED** [review]

**Comment by Zimmi48** [comment]
This PR was already backported as part of #11393.

**Pull request merged**
